packages/@aws-cdk/aws-events-targets/lib/ecs-task.ts

```diff
@@ -0,0 +1,166 @@
+   *
+   * (Only applicable in case the TaskDefinition is configured for AwsVpc networking)
+   *
+   * @default A new security group is created
+   */
+  readonly securityGroup?: ec2.ISecurityGroup;
+}
+
+/**
+ * Start a task on an ECS cluster
+ */
+export class EcsTask implements events.IRuleTarget {
+  public readonly securityGroup?: ec2.ISecurityGroup;
+  private readonly cluster: ecs.ICluster;
+  private readonly taskDefinition: ecs.TaskDefinition;
+  private readonly taskCount: number;
+
+  constructor(private readonly props: EcsTaskProps) {
+    this.cluster = props.cluster;
+    this.taskDefinition = props.taskDefinition;
+    this.taskCount = props.taskCount !== undefined ? props.taskCount : 1;
+
+    if (this.taskDefinition.networkMode === ecs.NetworkMode.AwsVpc) {
+      this.securityGroup = props.securityGroup || new ec2.SecurityGroup(this.taskDefinition, 'SecurityGroup', { vpc: this.props.cluster.vpc });
+    }
+  }
+
+  /**
+   * Allows using tasks as target of CloudWatch events
+   */
+  public bind(rule: events.IRule): events.RuleTargetProperties {
+    const policyStatements = [new iam.PolicyStatement()
+      .addAction('ecs:RunTask')
+      .addResource(this.taskDefinition.taskDefinitionArn)
+      .addCondition('ArnEquals', { "ecs:cluster": this.cluster.clusterArn })
+    ];
+
+    // If it so happens that a Task Execution Role was created for the TaskDefinition,
+    // then the CloudWatch Events Role must have permissions to pass it (otherwise it doesn't).
+    if (this.taskDefinition.executionRole !== undefined) {
+      policyStatements.push(new iam.PolicyStatement()
+        .addAction('iam:PassRole')
+        .addResource(this.taskDefinition.executionRole.roleArn));
+    }
+
+    // For Fargate task we need permission to pass the task role.
+    if (this.taskDefinition.isFargateCompatible) {
+      policyStatements.push(new iam.PolicyStatement()
+        .addAction('iam:PassRole')
+        .addResource(this.taskDefinition.taskRole.roleArn));
+    }
+
+    const id = this.taskDefinition.node.id + '-on-' + this.cluster.node.id;
+    const arn = this.cluster.clusterArn;
+    const role = singletonEventRole(this.taskDefinition, policyStatements);
+    const input = { containerOverrides: this.props.containerOverrides };
+    const taskCount = this.taskCount;
+    const taskDefinitionArn = this.taskDefinition.taskDefinitionArn;
+
+    // Use a custom resource to "enhance" the target with network configuration
+    // when using awsvpc network mode.
+    if (this.taskDefinition.networkMode === ecs.NetworkMode.AwsVpc) {
+      const subnetSelection = this.props.subnetSelection || { subnetType: ec2.SubnetType.Private };
+      const assignPublicIp = subnetSelection.subnetType === ec2.SubnetType.Private ? 'DISABLED' : 'ENABLED';
+
+      new cloudformation.AwsCustomResource(this.taskDefinition, 'PutTargets', {
+        onUpdate: { // We don't need an onDelete here because the target will be owned by CF anyway
```

Comment:
This is confusing to me. Why no `onCreate`?

Reply:
`onCreate` defaults to `onUpdate`

Reply:
https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-cloudformation/lib/aws-custom-resource.ts#L83

Reply:
Got it.

Reply:
I have to say I have mixed feelings about this implementation... but I guess this is the point of the CDK :)

Reply:
Which implementation, `AwsCustomResource` or this ECS target?

Reply:
Yes, the fact that it is using a custom resource to achieve its goal.